Lib/test/test_socket.py

```diff
@@ -6254,5 +6254,48 @@ def test_main():
     support.run_unittest(*tests)
     support.threading_cleanup(*thread_info)
 
+
+@requireAttrs(socket, "send_fds")
+@requireAttrs(socket, "recv_fds")
+@requireAttrs(socket, "AF_UNIX")
+class SendRecvFdsTests(unittest.TestCase):
```

Comment:
Oh, this test case must be added manually into test_main(). Currently, the test is not run...

I added it manually and... the test fails :-( The test writes into the read end of the pipe, and reads from the write end of the pipe. It should be the opposite. Maybe on FreeBSD, pipe ends are not directional, but it seems to be the case on Linux.

So the test should be fixed on Linux a well.

Please move this class before test_main().